iPetCare/Application/Services/InstitutionService.cs

```diff
@@ -0,0 +1,221 @@
+
+            return new ServiceResponse(HttpStatusCode.OK);
+        }
+
+        public async Task<ServiceResponse> SignUpAsync(Guid institutionId)
+        {
+            if (institutionId == Guid.Empty)
+                return new ServiceResponse(HttpStatusCode.NotFound, "Nieprawidłowy institutionId");
+
+            var username = CurrentlyLoggedUserName;
+            if (string.IsNullOrWhiteSpace(username))
+                return new ServiceResponse(HttpStatusCode.Unauthorized);
+
+            var user = await Context.Users.SingleOrDefaultAsync(u => u.UserName == username);
+            if (user == null)
+                return new ServiceResponse(HttpStatusCode.BadRequest, "Nie znaleziono użytkownika");
+
+            if (user.Role != Role.Vet)
+                return new ServiceResponse(HttpStatusCode.Forbidden);
+
+            var institution = Context.Institutions.Find(institutionId);
+
+            if(institution == null)
+                return new ServiceResponse(HttpStatusCode.NotFound, "Nie znaleziono instytucji");
+
+            var institutionVets = await  Context.InstitutionVets.Where(x => x.InstitutionId == institutionId && x.VetId == user.Vet.Id).ToListAsync();
+
+            if(institutionVets.Any())
+                return new ServiceResponse(HttpStatusCode.BadRequest, $"Weterynarz jest już zapisany do instytucji {institution.Name}");
+
+            var institutionVet = new InstitutionVet() {Vet = user.Vet, Institution = institution};
+            Context.Add(institutionVet);
+
+            if(await Context.SaveChangesAsync() <= 0)
+                return new ServiceResponse(HttpStatusCode.BadRequest, "Błąd podczas zapisu do bazy weterynarza do instytucji");
+
+            return new ServiceResponse(HttpStatusCode.OK);
+        }
+
+        public async Task<ServiceResponse> SignOutAsync(Guid institutionId)
+        {
+            if (institutionId == Guid.Empty)
+                return new ServiceResponse(HttpStatusCode.NotFound, "Nieprawidłowy institutionId");
+
+            var username = CurrentlyLoggedUserName;
+            if (string.IsNullOrWhiteSpace(username))
+                return new ServiceResponse(HttpStatusCode.Unauthorized);
+
+            var user = await Context.Users.SingleOrDefaultAsync(u => u.UserName == username);
+            if (user == null)
+                return new ServiceResponse(HttpStatusCode.BadRequest, "Nie znaleziono użytkownika");
+
+            if (user.Role != Role.Vet)
+                return new ServiceResponse(HttpStatusCode.Forbidden);
+
+            var institution = Context.Institutions.Find(institutionId);
+
+            if (institution == null)
+                return new ServiceResponse(HttpStatusCode.NotFound, "Nie znaleziono instytucji");
+
+            var institutionVets = await Context.InstitutionVets.Where(x => x.InstitutionId == institutionId && x.VetId == user.Vet.Id).ToListAsync();
```

Comment:
Zakładamy sytuację, że jeden vet będzie wiele razy przypisany do tej samej placówki? Chyba, że źle to rozumiem. Dlaczego RemoveRange poniżej? Wydaje mi się, że powinien być tylko jeden jedyny wpis w takiej tabeli. SingleOrDefault byłoby wtedy najlepszym wyborem

Reply:
A to moje 300 iq myślenie, chciałem żeby w razie ewentualności wywaliło zduplikowane połączenia XD. Poprawię na SingleOrDefault bo to sensowniejsze